File: ruby/005/main.rb
=begin
    ProjectEuler 005
    Ruby / main.rb
    copyright (c) 2015 Susisu

    Ruby 2.0.0p481
    `ruby main.rb`
=end

def main
    p = 1
    for i in 1 .. 20
        p *= i / gcd(p, i)
    end
    puts p
end

def gcd(a, b)
    if a < b then
        return gcd(b, a)
    elsif b == 0 then
        return a
    else
        return gcd(a % b, b)
    end
end

main()

File: perl/005/main.pl
# ProjectEuler 005
# Perl / main.pl
# copyright (c) 2015 Susisu
#
# Perl 5.16.2
# `perl main.pl`

sub gcd {
    my ($a, $b) = @_;
    if ($a < $b) {
        return gcd($b, $a);
    }
    elsif ($b == 0) {
        return $a;
    }
    else {
        return gcd($b, $a % $b);
    }
}

sub main {
    my $p = 1;
    for (my $i = 1; $i <= 20; $i++) {
        $p *= $i / gcd($p, $i);
    }
    print $p . "\n";
}

&main;
